logstreams/src/main/java/io/zeebe/logstreams/processor/StreamProcessorController.java

```diff
@@ -442,8 +443,13 @@ protected void onActorClosing() {
     metrics.close();
 
     if (!isFailed()) {
-      createSnapshot();
       streamProcessor.onClose();
+      createSnapshot();
+
+      final StateController stateController = streamProcessor.getStateController();
```

Comment:
I think it makes sense that we decouple this and close it after the stream processor.